src/commands.rs

```diff
@@ -662,6 +666,81 @@ pub fn json_set<M: Manager>(manager: M, ctx: &Context, args: Vec<RedisString>) -
     }
 }
 
+///
+/// JSON.MSET <key> <path> <json> [[<key> <path> <json>]...]
+///
+pub fn json_mset<M: Manager>(manager: M, ctx: &Context, args: Vec<RedisString>) -> RedisResult {
+    let mut args = args.into_iter().skip(1);
+
+    if args.len() < 3 {
+        return Err(RedisError::WrongArity);
+    }
+
+    // Collect all the actions from the args (redis_key, update_info, value)
+    let mut actions = Vec::new();
+    while let Ok(key) = args.next_arg() {
+        let mut redis_key = manager.open_key_write(ctx, key)?;
+
+        // Verify the key is a JSON type
+        let key_value = redis_key.get_value()?;
+
+        // Verify the path is valid and get all the update info
+        let path = Path::new(args.next_str()?);
+        let update_info = if path.get_path() == JSON_ROOT_PATH {
+            None
+        } else if let Some(value) = key_value {
+            Some(KeyValue::new(value).find_paths(path.get_path(), &SetOptions::None)?)
+        } else {
+            return Err(RedisError::Str(
+                "ERR new objects must be created at the root",
+            ));
+        };
+
+        // Parse the input and validate it's valid JSON
+        let value_str = args.next_str()?;
+        let value = manager.from_str(value_str, Format::JSON, true)?;
+
+        actions.push((redis_key, update_info, value));
+    }
+
+    actions
+        .drain(..)
+        .for_each(|(mut redis_key, update_info, value)| {
+            if let Some(mut update_info) = update_info {
+                if !update_info.is_empty() {
```

Comment:
Maybe we can extract this to a function somehow? Cause its the same as json.set no?

Reply:
maybe we can, it's not exactly the same, since in JSON.SET we also have the NX/EX flags

Reply:
done